pkg/api/scylla/v1alpha1/types_operatorconfig.go

```diff
@@ -25,6 +25,11 @@ type ScyllaOperatorConfigSpec struct {
 	// Setting this field renders your cluster unsupported. Use at your own risk.
 	// +optional
 	UnsupportedPrometheusVersionOverride *string `json:"unsupportedPrometheusVersionOverride,omitempty"`
+
+	// clusterDomain allows to set Kubernetes cluster domain used by the operator as a cluster identifier.
+	// It's set automatically discovered if not provided.
+	// +optional
+	ClusterDomain *string `json:"clusterDomain,omitempty"`
```

Comment:
I think having something that has a scheme (DNS) is useful. Knowing the cluster domain name may be helpful to communicate with service across cluster, if that becomes needed at some point (like shared manager or other shared services, multiDC, ...).

I can see that this may seem a bit misleading to some, although I'd not expect `ScyllaOperatorConfig.spec.clusterDomain` to configure the kubernetes DNS service itself.

a)
```suggestion
	// clusterDomain allows users to set the configured Kubernetes cluster domain explicitly, instead of letting the scylla operator to automatically discover it.
	// +optional
	ClusterDomain *string `json:"clusterDomain,omitempty"`
```

b)
```suggestion
	// configuredClusterDomain allows users to set the configured Kubernetes cluster domain explicitly, instead of letting the scylla operator to automatically discover it.
	// +optional
	ConfiguredClusterDomain *string `json:"configuredClusterDomain,omitempty"`
```

Reply:
> Knowing the cluster domain name may be helpful to communicate with services across clusters, if that becomes needed at some point

Why would the cluster domain be useful then? Doesn't the multicluster dns actually depend on the concept of a cluster identifier?

Reply:
The sole purpose of it is to identify from which Cluster resources are originating. Identifier doesn't need to be a domain, these usually has restrictions, but it could be anything human readable. We will use cluster domain as there's nothing better available, but users should be able to set their own string, not necessarily something that passes DNS name requirements.

Reply:
> The sole purpose of it is to identify from which Cluster resources are originating

yes, we don't need the domain specified for today's task, but we already go through the hussle of discovering it. I don't think it's restricting to limit this filed to a DNS label (label value is similarly restricted already) and we get the benefit of having an API field with the cluster domain, if we need it in the future - otherwise we then have to introduce a new field which will effectively deprecate this one as it will be a superset of it

Reply:
> Why would the cluster domain be useful then (or rather why assume it would be more useful than some other kind of identifier)? 

Say you want to talk to a manager / monitoring in a different cluster, or something in those lines - my main point is we already auto discover it, so let's just not degrade it by taking the scheme away.

Reply:
I agree, we will need cluster domain eventually, hence there's no point of having two fields for the same thing. Changed name and added validation.